gsms.py

```diff
@@ -153,6 +157,176 @@ def get_win_path(folder_id: str) -> str:
     # Free memory used by pointer
     _CoTaskMemFree(path_pointer)
 
+    print(path)
+
+    return path
+
+
+def copy_image(src_image: str, dst_image: str) -> None:
+    """
+    Copies an image from src_image to dst_image if the dst is empty or different
+
+    Parameters
+    ----------
+    src_image: str
+        Source path of the image
+    dst_image: str
+        Destination path of the image
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    >>> copy_image("C:\\Image1.png", "D:\\Image1.png")
+    """
+    # if src_image exists and dst_image does not exist
+    if os.path.isfile(src_image) and not os.path.isfile(dst_image):
+        shutil.copy2(src=src_image, dst=dst_image)  # copy2 preserves metadata
+        print(f'Copied box-art image to: {dst_image}')
+    else:
+        print(f'No box-art image found at: {src_image}')
+
+
+def has_app(sunshine_apps: any, name: str) -> bool:
```

Comment:
Firstly, `sunshine_apps` would be a `dict` type.

Secondly, since `sunshine_apps` is defined in `main`, you don't need to send this to the function. Dictionaries are are global objects. (Also applies to `add_game`).

Reply:
Updated in my most recent review.